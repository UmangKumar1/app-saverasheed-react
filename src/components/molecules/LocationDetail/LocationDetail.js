import React from 'react'
import styled, { css } from 'react-emotion'
import PropTypes from 'prop-types'

import ModalBasic from '../ModalBasic'

const { shape, string, func } = PropTypes

const PopupTitle = styled('h3')({
  textAlign: 'center',
  marginBlockStart: 0,
})
const PopupDescription = styled('div')({
  marginBottom: 10,
})
const PopupHint = styled('div')({
  fontStyle: 'italic',
})

const popupWrapperCx = css({
  bottom: 90,
})

const X = styled('div')({
  position: 'absolute',
  top: 15,
  right: 15,
})

const LocationDetail = ({
  closePopup,
  popupInfo: { name, description, hint },
}) => (
  <ModalBasic
    className="map-popup"
    popupWrapperClassName={popupWrapperCx}
    onClickAway={e => {
      e.preventDefault()
      e.stopPropagation()
      e.target.id !== 'deckgl-overlay' && closePopup()
    }}
  >
    <X onClick={closePopup}>X</X>
    <PopupTitle>{name}</PopupTitle>
    <PopupDescription>{description}</PopupDescription>
    <PopupHint>Hint: {hint}</PopupHint>
  </ModalBasic>
)

LocationDetail.propTypes = {
  closePopup: func.isRequired,
  popupInfo: shape({
    name: string,
    description: string,
    hint: string,
  }).isRequired,
}

export default LocationDetail
