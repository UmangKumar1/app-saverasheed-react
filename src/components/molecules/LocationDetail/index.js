export { default, default as LocationDetail } from './LocationDetail'
