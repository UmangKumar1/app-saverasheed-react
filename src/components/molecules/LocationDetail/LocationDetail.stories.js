import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import LocationDetail from '.'
import { poi1 } from '../../../fixtures/pois'

const popupInfo = {
  name: poi1.name,
  description: poi1.description,
  hint: poi1.hint,
}

storiesOf('LocationDetail', module).add('default', () => (
  <LocationDetail closePopup={action('close!')} popupInfo={popupInfo} />
))
