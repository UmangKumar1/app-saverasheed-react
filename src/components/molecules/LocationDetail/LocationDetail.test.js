import React from 'react'
import { shallow } from 'enzyme'
import LocationDetail from '.'
import { poi1 } from '../../../fixtures/pois'

const wrap = (props = {}) => shallow(<LocationDetail {...props} />)

const popupInfo = {
  name: poi1.name,
  description: poi1.description,
  hint: poi1.hint,
}

describe('LocationDetail', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      closePopup: jest.fn(),
      popupInfo,
    })
    expect(wrapper).toMatchSnapshot()
  })
})
