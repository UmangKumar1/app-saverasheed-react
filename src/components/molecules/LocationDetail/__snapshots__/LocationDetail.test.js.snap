// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationDetail renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LocationDetail
    closePopup={[MockFunction]}
    popupInfo={
      Object {
        "description": "Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?",
        "hint": "Take a stroll down the boardwalk",
        "name": "Broadway Pier",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Styled(div)
          onClick={[MockFunction]}
        >
          X
        </Styled(div)>,
        <Styled(h3)>
          Broadway Pier
        </Styled(h3)>,
        <Styled(div)>
          Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?
        </Styled(div)>,
        <Styled(div)>
          Hint: 
          Take a stroll down the boardwalk
        </Styled(div)>,
      ],
      "className": "map-popup",
      "onClickAway": [Function],
      "popupWrapperClassName": "css-11u3goc",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "X",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "X",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Broadway Pier",
        },
        "ref": null,
        "rendered": "Broadway Pier",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?",
        },
        "ref": null,
        "rendered": "Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "Hint: ",
            "Take a stroll down the boardwalk",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Hint: ",
          "Take a stroll down the boardwalk",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Styled(div)
            onClick={[MockFunction]}
          >
            X
          </Styled(div)>,
          <Styled(h3)>
            Broadway Pier
          </Styled(h3)>,
          <Styled(div)>
            Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?
          </Styled(div)>,
          <Styled(div)>
            Hint: 
            Take a stroll down the boardwalk
          </Styled(div)>,
        ],
        "className": "map-popup",
        "onClickAway": [Function],
        "popupWrapperClassName": "css-11u3goc",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "X",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "X",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Broadway Pier",
          },
          "ref": null,
          "rendered": "Broadway Pier",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?",
          },
          "ref": null,
          "rendered": "Look out over beautiful San Diego harbor, do you see Jason Mraz or Nick Cannon?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Hint: ",
              "Take a stroll down the boardwalk",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Hint: ",
            "Take a stroll down the boardwalk",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
